test/api/v3/integration/user/auth/POST-user_auth_social.test.js

```diff
@@ -22,38 +23,101 @@ describe('POST /user/auth/social', () => {
+      network,
     })).to.eventually.be.rejected.and.eql({
       code: 401,
       error: 'NotAuthorized',
-      message: t('onlyFbSupported'),
+      message: t('unsupportedNetwork'),
     });
   });
 
-  it('registers a new user', async () => {
-    let response = await api.post(endpoint, {
-      authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
-      network,
+  describe('facebook', () => {
+    before(async () => {
+      let expectedResult = {id: facebookId};
+      let passportFacebookProfile = sinon.stub(passport._strategies.facebook, 'userProfile');
+      passportFacebookProfile.yields(null, expectedResult);
+      network = 'facebook';
+    });
+
+    it('registers a new user', async () => {
+      let response = await api.post(endpoint, {
+        authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
+        network,
+      });
+
+      expect(response.apiToken).to.exist;
+      expect(response.id).to.exist;
+      expect(response.newUser).to.be.true;
     });
 
-    expect(response.apiToken).to.exist;
-    expect(response.id).to.exist;
-    expect(response.newUser).to.be.true;
+    it('logs an existing user in', async () => {
+      await user.update({ 'auth.facebook.id': facebookId });
+
+      let response = await api.post(endpoint, {
+        authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
+        network,
+      });
+
+      expect(response.apiToken).to.eql(user.apiToken);
+      expect(response.id).to.eql(user._id);
+      expect(response.newUser).to.be.false;
+    });
+
+    it('add social auth to an existing user', async () => {
+      let response = await user.post(endpoint, {
+        authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
+        network,
+      });
+
+      expect(response.apiToken).to.exist;
+      expect(response.id).to.exist;
+      expect(response.newUser).to.be.false;
+    });
   });
 
-  it('logs an existing user in', async () => {
-    await user.update({ 'auth.facebook.id': facebookId });
+  describe('google', () => {
+    before(async () => {
+      let expectedResult = {id: googleId};
+      let passportGoogleProfile = sinon.stub(passport._strategies.google, 'userProfile');
+      passportGoogleProfile.yields(null, expectedResult);
+      network = 'google';
+    });
+    it('registers a new user', async () => {
+      let response = await api.post(endpoint, {
+        authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
+        network,
+      });
 
-    let response = await api.post(endpoint, {
-      authResponse: {access_token: randomAccessToken}, // eslint-disable-line camelcase
-      network,
+      expect(response.apiToken).to.exist;
+      expect(response.id).to.exist;
+      expect(response.newUser).to.be.true;
+    });
+
+    it('logs an existing user in', async () => {
+      await user.update({ 'auth.facebook.id': facebookId });
```

Comment:
I think the line commit got shifted around since I made this comment. If I recall, it _was_ in the Google section. (or I am just completely mistaken)